homeassistant/components/mqtt/trigger.py

```diff
@@ -37,6 +37,17 @@ async def async_attach_trigger(hass, config, action, automation_info):
     encoding = config[CONF_ENCODING] or None
     qos = config[CONF_QOS]
     job = HassJob(action)
+    variables = None
+    if automation_info:
+        variables = automation_info.get("variables")
+
+    if payload:
+        template.attach(hass, payload)
+        payload = payload.async_render(variables, limited=True)
```

Comment:
We have a variable render helper that renders them 1 key at a time so that newer variables can reference already rendered ones.

Reply:
oh nevermind, this is the payload